core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -646,7 +645,17 @@ public ParseField getParseField() {
             return parseField;
         }
 
-        public ScriptMode getDefaultScriptMode() {
+        public ScriptMode getDefaultScriptMode(@Nullable String language) {
+            if (language != null) {
```

Comment:
Sounds good, I'll add an option to the interface so languages can specify this themselves.